istio/src/main/scala/io/buoyant/k8s/istio/mixer/MixerApiRequests.scala

```diff
@@ -0,0 +1,121 @@
+package io.buoyant.k8s.istio.mixer
+
+import com.fasterxml.jackson.databind.{DeserializationFeature, ObjectMapper}
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory
+import com.fasterxml.jackson.module.scala.DefaultScalaModule
+import com.fasterxml.jackson.module.scala.experimental.ScalaObjectMapper
+import com.google.local.DurationProto.{Duration => GDuration}
+import com.twitter.util.Duration
+import io.buoyant.k8s.istio._
+import istio.mixer.v1._
+
+import scala.io.Source
+
+object MixerApiRequests {
+
+  private[this] lazy val istioGlobalDict: Seq[String] = {
+    val yaml = Source.fromInputStream(
+      getClass.getClassLoader.getResourceAsStream("mixer/v1/global_dictionary.yaml")
+    ).mkString
+
+    val mapper = new ObjectMapper(new YAMLFactory) with ScalaObjectMapper
+    mapper.registerModule(DefaultScalaModule)
+    mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+
+    mapper.readValue[Seq[String]](yaml)
+  }
+
+  def mkReportRequest(
+    responseCode: ResponseCodeIstioAttribute,
+    requestPath: RequestPathIstioAttribute,
+    targetService: TargetServiceIstioAttribute,
+    sourceLabel: SourceLabelIstioAttribute,
+    targetLabel: TargetLabelsIstioAttribute,
+    duration: ResponseDurationIstioAttribute
+  ): ReportRequest = {
+
+    val dictionary = mkDictionary(requestPath, targetService, sourceLabel, targetLabel)
+
+    val updated = mkAttributesUpdate(dictionary, Seq(responseCode, requestPath, targetService, sourceLabel, targetLabel, duration))
+    ReportRequest(attributeUpdate = Some(updated))
+  }
+
+  def mkCheckRequest(istioREquest: IstioRequest[_]) = {
+    val TODO = Some(1L)
+    val dictionary = mkDictionary(istioREquest.requestedPath, istioREquest.targetService, istioREquest.sourceLabel, istioREquest.targetLabel)
+    val requestAttributes = Seq(istioREquest.requestedPath, istioREquest.sourceLabel, istioREquest.targetLabel, istioREquest.targetService)
+    val attributesUpdate = mkAttributesUpdate(dictionary, requestAttributes)
+    CheckRequest(TODO, Some(attributesUpdate))
+  }
+
+  private def mkDictionary(
```

Comment:
This whole function seems really convoluted, is there any way it can be simplified or broken apart?

Reply:
I think the current version might be a little bit better? https://github.com/linkerd/linkerd/pull/1650/files#diff-00e114bad2c9fdd832ebe5854d4a1a4fR50